.github/workflows/update_file.yaml

```diff
@@ -0,0 +1,47 @@
+# This workflow is to sync dashboards and alert rules from the Sunbeam repo and create a PR if there is an update.
+name: Update files
+on:
+  workflow_dispatch:
+  workflow_call:
+  schedule:
+    - cron: '43 1 * * 1'
+
+jobs:
+
+  pull-from-sunbeam:
+    name: Pull files from sunbeam
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout local repository
+        uses: actions/checkout@v4
+
+      - name: set git config
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          git config --global user.email "${GITHUB_ACTOR_ID}+${GITHUB_ACTOR}@users.noreply.github.com"
+          git config --global user.name "$(gh api /users/${GITHUB_ACTOR} | jq .name -r)"
+          git config -l
+          
+      - name: Pull dashboard and alert rules from sunbeam
+        run: |
+          git clone --sparse https://opendev.org/openstack/sunbeam-charms.git
+          cd sunbeam-charms
+          git sparse-checkout set charms/openstack-exporter-k8s/src/grafana_dashboards charms/openstack-exporter-k8s/src/prometheus_alert_rules
+          cp -r charms/openstack-exporter-k8s/src/grafana_dashboards ../src/
+          cp -r charms/openstack-exporter-k8s/src/prometheus_alert_rules ../src/
+          cd ..
+          rm -rf sunbeam-charms
+
+      - name: Create a PR for local changes
+        uses: peter-evans/create-pull-request@v6
+        id: cpr
+        with:
+          commit-message: "chore: update dashboards and alert rules"
+          title: "chore: update dashboards and alert rules"
+          body: |
+            Automated action to fetch the latest grafana dashboards and alert rules from Sunbeam. The branch of this PR 
+            will be overwritten during the next check. Unless you really know what you're doing, you 
+            most likely don't want to push any commits to this branch.
```

Comment:
To assist reviewers, it would be helpful to add links to the relevant directories on https://opendev.org/openstack/sunbeam-charms for reference.